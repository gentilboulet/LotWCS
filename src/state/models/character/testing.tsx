import { applyMiddleware, compose, createStore } from "redux";
import { actions as characterActions } from "../../actions/character";

import {
  globalReducer,
  IAction,
  initialStateFactory,
  IStoreState,
} from "../../../state";

import { middleware as applyPerks } from "../../../state/middleware/character/applyPerks";
import { middleware as checkAutomatics } from "../../../state/middleware/character/automatics";
import { middleware as payCosts } from "../../../state/middleware/character/payCosts";
import { middleware as pushToHistory } from "../../../state/middleware/pushToHistory";

import { zeroCost } from "./costs";

export const testingStateFactory = () => {
  const actions = [
    characterActions.setName("Testing state"),
    characterActions.setRank(2),
    characterActions.skillsBuy("Awareness", zeroCost),
    characterActions.skillSpecialityBuy(
      "Confidence",
      "Appear steadfast",
      zeroCost,
    ),
    characterActions.skillSpecialityBuy(
      "Confidence",
      "Another speciality",
      zeroCost,
    ),
    characterActions.armorBuy(
      "heavy han armor",
      "Heavy armor worn by soldiers of the Forbidden City",
      "Heavy Armor",
      zeroCost,
    ),
    characterActions.weaponBuy(
      "Flexible simple weapon",
      "Flexible snake sword",
      ["Massive"],
      zeroCost,
    ),
    characterActions.weaponBuy(
      "Horse splitter",
      "A sword so big it could chop horses in one cleave",
      ["Massive", "Sword"],
      zeroCost,
    ),
  ];

  const store = createStore<IStoreState, IAction, any, any>(
    globalReducer,
    initialStateFactory(),
    compose(
      applyMiddleware(payCosts),
      applyMiddleware(checkAutomatics),
      applyMiddleware(pushToHistory),
      applyMiddleware(applyPerks),
    ),
  );

  actions.forEach(a => {
    store.dispatch(a);
  });

  return store.getState().character;
};
