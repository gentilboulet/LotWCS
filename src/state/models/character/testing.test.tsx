import { testingStateFactory } from "./testing";

describe("testing state", () => {
  it("should be stable", () => {
    expect(testingStateFactory()).toMatchSnapshot();
  });
});
