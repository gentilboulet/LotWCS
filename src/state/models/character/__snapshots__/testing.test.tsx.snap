// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing state should be stable 1`] = `
Object {
  "archetype": undefined,
  "automatics": Array [],
  "chi": Object {
    "corrupt": Object {
      "cultivation": 0,
      "value": 0,
    },
    "earth": Object {
      "cultivation": 0,
      "value": 0,
    },
    "enlightened": Object {
      "cultivation": 0,
      "value": 0,
    },
    "fire": Object {
      "cultivation": 0,
      "value": 0,
    },
    "general": Object {
      "cultivation": 0,
      "value": 10,
    },
    "metal": Object {
      "cultivation": 0,
      "value": 0,
    },
    "water": Object {
      "cultivation": 0,
      "value": 0,
    },
    "wood": Object {
      "cultivation": 0,
      "value": 0,
    },
  },
  "concept": undefined,
  "destiny": 20,
  "discounts": Array [
    Object {
      "skills": Array [
        "Awareness",
        "Confidence",
        "Crafting",
        "Finesse",
        "Hardiness",
        "Inspire",
        "Learning",
        "Medicine",
        "Might",
        "Perform",
        "Politics",
        "Ride",
        "Stealth",
        "Survival",
        "Tactics",
        "Wu Wei",
      ],
      "type": "DISCOUNT_SKILL",
      "value": 20,
    },
    Object {
      "kfType": "KUNGFU_EXTERNAL",
      "type": "DISCOUNT_KUNGFU_STYLE",
      "uids": Array [],
      "value": 10,
    },
    Object {
      "kfType": "KUNGFU_INTERNAL",
      "type": "DISCOUNT_KUNGFU_STYLE",
      "uids": Array [],
      "value": 10,
    },
    Object {
      "type": "DISCOUNT_VIRTUE",
      "value": 75,
      "virtues": Array [
        Object {
          "name": "Honor",
          "type": "VIRTUE_CHIVALROUS",
        },
        Object {
          "name": "Benevolence",
          "type": "VIRTUE_CHIVALROUS",
        },
        Object {
          "name": "Righteousness",
          "type": "VIRTUE_CHIVALROUS",
        },
        Object {
          "name": "Loyalty",
          "type": "VIRTUE_CHIVALROUS",
        },
        Object {
          "name": "Force",
          "type": "VIRTUE_CHIVALROUS",
        },
        Object {
          "name": "Revenge",
          "type": "VIRTUE_SELFISH",
        },
        Object {
          "name": "Individualism",
          "type": "VIRTUE_SELFISH",
        },
        Object {
          "name": "Obsession",
          "type": "VIRTUE_SELFISH",
        },
        Object {
          "name": "Ruthlessness",
          "type": "VIRTUE_SELFISH",
        },
        Object {
          "name": "Ferocity",
          "type": "VIRTUE_SELFISH",
        },
      ],
    },
  ],
  "entanglement": 0,
  "gear": Object {
    "armors": Array [
      Object {
        "description": "Heavy armor worn by soldiers of the Forbidden City",
        "name": "heavy han armor",
        "type": "Heavy Armor",
      },
    ],
    "weapons": Array [
      Object {
        "description": "Flexible snake sword",
        "name": "Flexible simple weapon",
        "type": Array [
          "Massive",
        ],
      },
      Object {
        "description": "A sword so big it could chop horses in one cleave",
        "name": "Horse splitter",
        "type": Array [
          "Massive",
          "Sword",
        ],
      },
    ],
  },
  "kungfu": Object {
    "KUNGFU_EXTERNAL": Object {},
    "KUNGFU_INTERNAL": Object {},
    "noRestrictionInternal": Array [],
  },
  "loresheets": Object {},
  "name": "Testing state",
  "rank": 2,
  "skills": Array [
    Object {
      "name": "Awareness",
      "specialities": Array [],
      "value": 5,
    },
    Object {
      "name": "Confidence",
      "specialities": Array [
        "Appear steadfast",
        "Another speciality",
      ],
      "value": 0,
    },
    Object {
      "name": "Crafting",
      "specialities": Array [],
      "value": 0,
    },
    Object {
      "name": "Finesse",
      "specialities": Array [],
      "value": 0,
    },
    Object {
      "name": "Hardiness",
      "specialities": Array [],
      "value": 0,
    },
    Object {
      "name": "Inspire",
      "specialities": Array [],
      "value": 0,
    },
    Object {
      "name": "Learning",
      "specialities": Array [],
      "value": 0,
    },
    Object {
      "name": "Medicine",
      "specialities": Array [],
      "value": 0,
    },
    Object {
      "name": "Might",
      "specialities": Array [],
      "value": 0,
    },
    Object {
      "name": "Perform",
      "specialities": Array [],
      "value": 0,
    },
    Object {
      "name": "Politics",
      "specialities": Array [],
      "value": 0,
    },
    Object {
      "name": "Ride",
      "specialities": Array [],
      "value": 0,
    },
    Object {
      "name": "Stealth",
      "specialities": Array [],
      "value": 0,
    },
    Object {
      "name": "Survival",
      "specialities": Array [],
      "value": 0,
    },
    Object {
      "name": "Tactics",
      "specialities": Array [],
      "value": 0,
    },
    Object {
      "name": "Wu Wei",
      "specialities": Array [],
      "value": 0,
    },
  ],
  "virtues": Array [
    Object {
      "name": "Honor",
      "type": "VIRTUE_CHIVALROUS",
      "value": 0,
    },
    Object {
      "name": "Benevolence",
      "type": "VIRTUE_CHIVALROUS",
      "value": 0,
    },
    Object {
      "name": "Righteousness",
      "type": "VIRTUE_CHIVALROUS",
      "value": 0,
    },
    Object {
      "name": "Loyalty",
      "type": "VIRTUE_CHIVALROUS",
      "value": 0,
    },
    Object {
      "name": "Force",
      "type": "VIRTUE_CHIVALROUS",
      "value": 0,
    },
    Object {
      "name": "Revenge",
      "type": "VIRTUE_SELFISH",
      "value": 0,
    },
    Object {
      "name": "Individualism",
      "type": "VIRTUE_SELFISH",
      "value": 0,
    },
    Object {
      "name": "Obsession",
      "type": "VIRTUE_SELFISH",
      "value": 0,
    },
    Object {
      "name": "Ruthlessness",
      "type": "VIRTUE_SELFISH",
      "value": 0,
    },
    Object {
      "name": "Ferocity",
      "type": "VIRTUE_SELFISH",
      "value": 0,
    },
  ],
}
`;
